rig-core/src/loaders/pdf.rs

```diff
@@ -0,0 +1,447 @@
+use std::{fs, path::PathBuf};
+
+use glob::glob;
+use lopdf::{Document, Error as LopdfError};
+use thiserror::Error;
+
+use super::file::FileLoaderError;
+
+#[derive(Error, Debug)]
+pub enum PdfLoaderError {
+    #[error("{0}")]
+    FileLoaderError(#[from] FileLoaderError),
+
+    #[error("UTF-8 conversion error: {0}")]
+    FromUtf8Error(#[from] std::string::FromUtf8Error),
+
+    #[error("IO error: {0}")]
+    PdfError(#[from] LopdfError),
+}
+
+trait Loadable {
+    fn load(self) -> Result<Document, PdfLoaderError>;
+    fn load_with_path(self) -> Result<(PathBuf, Document), PdfLoaderError>;
+}
```

Comment:
Make this trait private just like the `Readable` trait in the `FileLoader` module?

Reply:
Actually, I wanted `Readable` to be private but only to the crate since it's not relevant outside of `loaders`. I'm not sure how exactly to do that though.

Reply:
Yeah sorry for the abuse of language, when I said "private" I meant `pub(crate)`, which does indeed make the thing only public within the crate!

In fact, you could be even more granular and make it `pub(in crate::loaders)`, which would make them public only in the `loaders` module, but that might be overkill

Reply:
Oh, that's really nice to know. I did not know visibility was so granular!